crates/precompile/build.rs

```diff
@@ -0,0 +1,183 @@
+#![allow(dead_code, unused_imports)]
+
+use std::ffi::CString;
+use std::fs;
+use std::path::{Path, PathBuf};
+use std::slice;
+
+const BYTES_PER_G1_POINT: usize = 48;
+const BYTES_PER_G2_POINT: usize = 96;
+
+fn main() {
+    generate_kzg_settings();
+}
+
+fn generate_kzg_settings() {
+    // Note: we don't use `OUT_DIR` because we don't ship this build script with the crate, so all
+    // used files must be in tree.
+    // let out_dir = PathBuf::from(std::env::var("OUT_DIR").unwrap());
+    let out_dir = Path::new("src/blob/kzg_settings");
+    let out_path = out_dir.join("generated.rs");
+
+    let in_path = Path::new("src/blob/kzg_settings/trusted_setup.txt");
+    println!("cargo:rerun-if-changed={}", in_path.display());
+    assert!(in_path.exists());
+    let contents = format_kzg_settings_small(in_path, out_dir);
+    fs::write(out_path, contents).unwrap();
+}
+
+/// Pros:
+/// - smaller runtime static size (198K = `4096*48 + 65*96`)
+/// - smaller generated file size (~~720K~~ same as runtime size by using `.bin` files)
+/// - half the size of `trusted_setup.txt`, which would have to be included in the binary without
+///   this build script
+///
+/// Cons:
+/// - must call `load_trusted_setup` at least once
+fn format_kzg_settings_small(in_path: &Path, out_dir: &Path) -> String {
+    let contents = fs::read_to_string(in_path).unwrap();
+    let mut lines = contents.lines();
+
+    // load number of points
+    let n_g1 = lines.next().unwrap().parse::<usize>().unwrap();
+    let n_g2 = lines.next().unwrap().parse::<usize>().unwrap();
+
+    assert_eq!(n_g2, 65);
+
+    // TODO: We can use `hex_literal` to reduce file size
+    // load g1 points
+    let mut g1_points = Vec::with_capacity(n_g1);
+    for _ in 0..n_g1 {
+        let line = lines.next().unwrap();
+        let mut bytes = [0; BYTES_PER_G1_POINT];
+        hex::decode_to_slice(line, &mut bytes).unwrap();
+        g1_points.push(bytes);
+    }
+
+    // load g2 points
+    let mut g2_points = Vec::with_capacity(n_g2);
+    for _ in 0..n_g2 {
+        let line = lines.next().unwrap();
+        let mut bytes = [0; BYTES_PER_G2_POINT];
+        hex::decode_to_slice(line, &mut bytes).unwrap();
+        g2_points.push(bytes);
+    }
```

Comment:
Nit but as discussed you don't need to read all the points right?

Reply:
It's done at compile-time. There's only 2 methods exposed to init a `KzgSettings`, with the points bytes, or with a file path.